web3/hardhat/contracts/Crowdfund.sol

```diff
@@ -0,0 +1,134 @@
+//SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0 <0.9.0;
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+error NotEnoughETHDonation();
+error NotEnoughBalance();
+error FailedToSendNativeToken();
+
+contract Crowdfund is Ownable, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+
+    IERC20 public token;
+    uint256 public startsAt;
+    uint256 public endsAt;
+    uint256 public goal;
+    bytes public metaPtr;
+
+    bool public canRefund;
+
+    mapping(address => uint256) userFunds;
+
+    event donated(address sender, uint256 amount);
+    event funded(address to, uint256 amount);
+    event refundActive(bool refunding);
+    event userBalanceWithdrawn(address user, uint256 balance);
+
+    modifier crowfundingEnded() {
+        require(!hasCrowdfundingEnded(), "Crowd funding hasn't ended");
+        _;
+    }
+
+    constructor(bytes memory meta) {
+        // pass data to constructor for new matchingPool
+        (
+            address safe,
```

Comment:
Rename to owner? Or is there any way of checking if an address has the Safe interface?
And consistency in names with a _ prefix